_Week9/Group4/estimation.do

```diff
@@ -0,0 +1,99 @@
+********************************************************************************
+* ECON 491
+* Week 9 - Estimation
+* Group 4
+* November 14, 2022
+********************************************************************************
+
+cap prog drop projectdata_sim
+
+prog define projectdata_sim, rclass
+
+clear
+
+********************************************************************************
+* setting up inputs
+********************************************************************************
+
+{
+	
+syntax anything
+local bias: word 1 of `anything' //a dummy for whether the results are biased or not
+local cluster_size: word 2 of `anything' //inputting cluster size
+local number_of_schools: word 3 of `anything' //inputting number of schools
+local te: word 4 of `anything' //inputting treatment effect
+local takeup_rate: word 5 of `anything' //inputting takeup rate
+
+}
+
+********************************************************************************
+* data generating process
+********************************************************************************
+
+{
+
+clear
+set obs `number_of_schools' //setting number of schools based on input
+generate school_id = _n
+generate cluster_size = `cluster_size' //setting number of students to be observed at each school
+gen random = runiform(0,1)
+sort random
+gen random_id = _n
+gen treat = 0 //randomly assigning treatment to half of schools
+replace treat = 1 if random_id > `number_of_schools'/2
+drop random random_id
+expand cluster_size //expand
+sort school_id
+generate student_id = _n
+gen random = runiform(0,1)
+gen takeup = 0 //randomly assigning takeup status
+replace takeup = 1 if random <= `takeup_rate'
```

Comment:
My primary piece of feedback is: in "real life," do you think it's realistic that take-up would be random, or would it be the result of some observable/unobservable characteristic of the school that also determines the effect of the treatment (and therefore your final estimate for the mental health score)? I could see a world where schools that don't take up the treatment also are less committed to their students more generally and provide a poorer education on average, which might further depress their mental health score. I think if this is the case, your estimate even if you take take-up into account as you do in your last section of the program would be biased. (The solution I've learned in this case is to use instrumental variables, which I think [this blog post](http://econometricsense.blogspot.com/2019/04/intent-to-treat-instrumental-variables.html) does a pretty good job of laying out, but I think the instructors would be able to discuss best why the estimate would be biased and how to fix it.

If you do think it would be truly random, then I think your code is great!